### PR TITLE
Vtcca 3203 simple page

### PR DESCRIPTION
**Pull request by reddy-yattapu**

### CODE REVIEW

File: app/assets/stylesheets/search.scss [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  -moz-box-sizing: content-box;
+  -webkit-box-sizing: content-box;
+  box-sizing: content-box;
+}
+
+@media (min-width: 641px) {
+  main.search .search-header input {
+    line-height: 1.31579;
+  }
+}
```
Comment (darrell-plant) [inline app/assets/stylesheets/search.scss:45]: Why is this file duplicated? 
app/assets/css/search.scss   and   app/assets/stylesheets/search.scss

File: app/controllers/agent/forms/FilterAgentProperties.scala [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import binders.propertylinks.GetPropertyLinksParameters
+
+case class FilterAgentProperties(address: Option[String], localAuthorityReference: Option[String]) {
+
+  def searchCriteriaExists: Boolean =
+    if (address.isDefined || localAuthorityReference.isDefined)
+      true
+    else
+      false
+}
```
Comment (darrell-plant) [inline app/controllers/agent/forms/FilterAgentProperties.scala:28]: def searchCriteriaExists: Boolean = address.isDefined || localAuthorityReference.isDefined

File: app/controllers/agent/ManageAgentController.scala [hunk 3]
```diff
@@ -75,6 +82,51 @@ class ManageAgentController @Inject()(
     }
   }
 
+  def manageAgentProperties(
+        agentCode: Long,
+        params: GetPropertyLinksParameters,
+        pagination: PaginationParameters): Action[AnyContent] = authenticated.async { implicit request =>
```
Comment (darrell-plant) [inline app/controllers/agent/ManageAgentController.scala:88]: pagination is not used

File: app/views/propertyrepresentation/manage/manageAgent.scala.html [hunk 4]
```diff
@@ -41,7 +42,7 @@
 
 @mainLayout(
     pageTitle = messages("propertyRepresentation.manageAgent.title"),
-    backUri = Some("#")
+    backUri = Some(controllers.agent.routes.ManageAgentController.manageAgentProperties(agent.representativeCode, GetPropertyLinksParameters(), PaginationParameters(1, 15)).url)
```
Comment (darrell-plant) [inline app/views/propertyrepresentation/manage/manageAgent.scala.html:45]: why PaginationParameters(1, 15) ?

File: conf/messages [hunk 5]
```diff
@@ -454,12 +460,15 @@ error.propertyRepresentation.agentAlreadyAppointed=This agent has already been a
 
 #Show your agents
 propertyRepresentation.showAgents.title=Your agents
+propertyRepresentation.agentProperties.assigned=Assigned properties
+propertyRepresentation.manage.agent=Manage this agent
 propertyRepresentation.showAgents.properties={0} of {1} properties
 propertyRepresentation.showAgents.viewAgentDProperties=<a class="govuk-link" href="{0}">View</a>
+error.propertyRepresentation.agentProperties.filter=Search Properties - you must enter something to search for
+error.agentProperties.filter=You must enter either address or Local authority reference something to search for
```
Comment (darrell-plant) [inline conf/messages:468]: Maybe check the text. The second one is not acceptable english. I'm not sure about the first. Better check.

File: test/connectors/PropertyLinkConnectorSpec.scala [hunk 6]
```diff
@@ -103,6 +108,19 @@ class PropertyLinkConnectorSpec extends VoaPropertyLinkingSpec {
     whenReady(connector.getMyOrganisationAgents())(_ mustBe organisationsAgentsList)
   }
 
+  "getMyAgentPropertyLinks" must "return agent property links" in new Setup {
+    val emptySearchParams = GetPropertyLinksParameters()
+    val paginationParams = PaginationParams(1, 1, true)
```
Comment (darrell-plant) [inline test/connectors/PropertyLinkConnectorSpec.scala:113]: PaginationParams(1, 15, true) would be better

Comment (darrell-plant) [reply test/connectors/PropertyLinkConnectorSpec.scala:113]: actually PaginationParams() defaults to good values doesn't it?

File: test/services/AgentRelationshipServiceSpec.scala [hunk 7]
```diff
@@ -54,6 +56,26 @@ class AgentRelationshipServiceSpec extends ServiceSpec with AllMocks {
 
   implicit val hc = HeaderCarrier(sessionId = Some(SessionId("1111")))
 
+  "getMyAgentPropertyLinks" should "return OwnerAuthResult when successful" in {
+    when(mockPropertyLinkConnector.getMyAgentPropertyLinks(any(), any(), any())(any()))
+      .thenReturn(Future.successful(ownerAuthResultWithTwoAuthorisation))
+    val res = testService.getMyAgentPropertyLinks(1, GetPropertyLinksParameters(), PaginationParams(1, 15, true))
```
Comment (darrell-plant) [inline test/services/AgentRelationshipServiceSpec.scala:62]: DefaultPaginationParams

File: app/controllers/agent/ManageAgentController.scala (hunk 3 above)
Comment (reddy-yattapu) [reply app/controllers/agent/ManageAgentController.scala:88]: Last week discussions with Karren, she wanted gohead with Search Functionality on Manage Agents but she chnaged her mind after yesterday our chat.

She wanted to display Simple Agents Property Details without "Search Functionality".

She will let me know soon when she wants swithcing between "Simple Agents" page and  "Search Functionality"

I will apply using Pagination and Filtering parameters when "Search Functionality" in play

File: app/views/propertyrepresentation/manage/manageAgent.scala.html (hunk 4 above)
Comment (reddy-yattapu) [reply app/views/propertyrepresentation/manage/manageAgent.scala.html:45]: That is default page size on Page Load When Search Functionality Applies
Page load from Back button, we need to reload the page with Default Pagination and Filter Parameters

Comment (darrell-plant) [reply app/views/propertyrepresentation/manage/manageAgent.scala.html:45]: ok - you can use DefaultPaginationParameters

File: test/connectors/PropertyLinkConnectorSpec.scala (hunk 6 above)
Comment (darrell-plant) [reply test/connectors/PropertyLinkConnectorSpec.scala:113]: DefaultPaginationParameters

File: conf/messages (hunk 5 above)
Comment (reddy-yattapu) [reply conf/messages:468]: I have edited, I have changed to same as one in Your Properties screen now and I will check with Karren about it When Search Functionality applies

File: test/connectors/PropertyLinkConnectorSpec.scala (hunk 6 above)
Comment (reddy-yattapu) [reply test/connectors/PropertyLinkConnectorSpec.scala:113]: Updated to use DefaultPaginationParams

File: test/services/AgentRelationshipServiceSpec.scala (hunk 7 above)
Comment (reddy-yattapu) [reply test/services/AgentRelationshipServiceSpec.scala:62]: Updated to use DefaultPaginationParams

File: app/controllers/agent/forms/FilterAgentProperties.scala (hunk 2 above)
Comment (reddy-yattapu) [reply app/controllers/agent/forms/FilterAgentProperties.scala]: updated

File: conf/messages [hunk 8]
```diff
@@ -465,7 +465,7 @@ propertyRepresentation.manage.agent=Manage this agent
 propertyRepresentation.showAgents.properties={0} of {1} properties
 propertyRepresentation.showAgents.viewAgentDProperties=<a class="govuk-link" href="{0}">View</a>
 error.propertyRepresentation.agentProperties.filter=Search Properties - you must enter something to search for
-error.agentProperties.filter=You must enter either address or Local authority reference something to search for
+error.agentProperties.filter=you must enter something to search for
```
Comment (darrell-plant) [inline conf/messages:468]: "You", not "you"

**Comment by darrell-plant** [comment]
Minor comment, then lgtm 👍

**Comment by jon-pike** [comment]
Seems to be a couple of Darrell's points that haven't been looked at. DefaultPaginationParams on manageAgent.scala.html and a content change in messages. Other than that 👍

**Pull request merged**